### PR TITLE
Include paths on pull_request event trigger for compile-queries.yml workflow

### PR DESCRIPTION
**Pull request by oscarsj**
Avoid running `compile-queries.yml` workflow on changes unrelated to queries (in particular on Kotlin extractor changes).

### CODE REVIEW

File: .github/workflows/compile-queries.yml [hunk 1]
```diff
@@ -7,6 +7,11 @@ on:
       - "rc/*"
       - "codeql-cli-*"
   pull_request:
+    paths:
+      - '*.ql'
+      - '*.qll'
+      - '**/qlpack.yml'
+      - '*.dbscheme'
```
Comment (igfoo) [inline .github/workflows/compile-queries.yml:14]: Don't all the lines need a `**/`?

Comment (oscarsj) [reply .github/workflows/compile-queries.yml]: Indeed, thanks for catch. I updated according to https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#example-including-paths

**Comment by criemen** [comment]
Is this correct? I do believe our qlucie-setup relies on compile-queries being run for external contributions.

**Comment by oscarsj** [comment]
> Is this correct? I do believe our qlucie-setup relies on compile-queries being run for external contributions.

Good catch, indeed it does. It does rely also on QL-for-QL workflow (`.github/workflows/ql-for-ql-build.yml`) that has the following triggers:
```
on:
  push:
    branches: [main]
  pull_request:
    branches: [main]

```
if I'm correct, it should be triggered for all PRs with a target branch `main`, so we wouldn't miss the `compile-queries.yml` worfklow, qlucie would still trigger the external checks.

**Review by criemen — APPROVED** [review]
TIL that `**.ql` on actions is a shorthand for `**/*.ql`. LGTM!

**Pull request merged**